frontend/src/Config/Attribution.elm

```diff
@@ -45,3 +45,18 @@ defaultGenre language =
 
         SP ->
             Article
+
+{-| Update Genre when switch languages
+* When switching languages, some genres are not available. For example, genre "email" only existed in English.
+- This function will keep set genre of new language same to previous language.
+- If genre is not existed in new language, this function will set it to default genre.
+-}
+changeGenre : Language -> Genre -> Genre
+changeGenre language genre =
+    case language of
```

Comment:
when seeing it like this, it can be slightly smaller still (nested cases are best avoided). 
```elm
case (language, genre) of 
    (EN, Essay) -> 
        -- explain why this case is special
        Essay

    _ -> defaultGenre language
```